README.md

```diff
@@ -1 +1,523 @@
+### 3.1.5 Memory Quota
+
+The OPTIONAL `memory` field configures the upper limit in system memory that the executor SHOULD allow.
+
+### 3.1.6 Disk Quota
+
+The OPTIONAL `disk` field configures the upper limit in system memory that the executor SHOULD allow.
+
+### 3.1.7 Gas Quota
+
+The OPTIONAL `gas` field configures the upper limit in Wasm gas that the executor SHOULD allow.
+
+## 3.2 IPLD Schema
+
+``` ipldsch
+type SystemConfig struct {
+  secret Boolean      (implicit False)
+  check  Verification (implicit Attestation)
+  gas    Integer      (implicit 0)
+  time   optional TimeInterval
+  memory optional InfoSize
+  disk   optional InfoSize
+}
+```
+
+## 3.3 JSON Examples
+
+``` json
+{
+  "secret": true,
+  "check": {
+    "optimistic": {
+      "confirmations": 2, 
+      "referee": "did:key:zStEZpzSMtTt9k2vszgvCwF4fLQQSyA15W5AQ4z3AR6Bx4eFJ5crJFbuGxKmbma4"
+    }
+  },
+  "gas": 5000,
+  "time": [45, "minutes"],
+  "memory": [500, "kilo", "bytes"],
+  "disk": [20, "mega", "bytes"]
+}
+```
+
+# 4 Task Configuration
+
+> With hands of iron, there's not a task we couldn't do
+>
+> — [The Protomen](https://en.wikipedia.org/wiki/The_Protomen), The Good Doctor
+
+Tasks are the smallest level of work granularity a workflow. Tasks describe everything required to the negotiate and execute all of the of work. IPVM Tasks are defined as a subtype of [UCAN Tasks](https://github.com/ucan-wg/invocation/blob/main/README.md#32-ipld-schema). Task types MAY require specific fields in the `inputs` field.  Timeouts, gas, credits, transactional guarantees, result visibility, and so on MAY be separately configured in the `ipvm/config` field.
+
+Tasks MAY be configured in aggragate in the [global defaults](#215-defaults). Individual Task configuration MUST be embedded inside of a [UCAN Action's](https://github.com/ucan-wg/invocation) `meta['ipvm/confg']` field.
+
+## 4.1 Fields
+
+Recall UCAN Invocation Tasks:
+
+| Field    | Type             | Description                                    | Required | Default |
+|----------|------------------|------------------------------------------------|----------|---------|
+| `with`   | `URI`            |                                                | Yes      |         |
+| `do`     | `Ability`        |                                                | Yes      |         |
+| `inputs` | `Any`            |                                                | Yes      |         |
+| `meta`   | `{String : Any}` | Fields that will be ignored during memoization | No       | `{}`    |
+
+An OPTIONAL IPVM `Config` MAY be included at the `meta['ipvm/config']` path. The `meta` field SHOULD not captured as part of task memoization, so this information will be omitted from the distributed invocation table. If included, the `Config` MUST set the IPVM configuration for this Task, overwriting any of the fields on the envelope's top-level `defaults` field, or system-wide defaults.
+
+## 4.2 JSON Examples
+
+``` js
+{
+  "some-wasm": {
```

Comment:
Note that these are shown in `{String:Task}` form – would it be clearer to just show the `Task`, to avoid confusion that the string is part of that?